scylla/src/frame/value.rs

```diff
@@ -218,6 +227,23 @@ impl<T: Value> Value for Option<T> {
     }
 }
 
+impl Value for Unset {
+    fn serialize(&self, buf: &mut Vec<u8>) -> Result<(), ValueTooBig> {
+        // Unset serializes itself to empty value with length = -2
+        buf.put_i32(-2);
+        Ok(())
+    }
+}
+
+impl<V: Value> Value for MaybeUnset<V> {
+    fn serialize(&self, buf: &mut Vec<u8>) -> Result<(), ValueTooBig> {
+        match self {
+            MaybeUnset::Set(v) => v.serialize(buf),
+            MaybeUnset::Unset => Unset {}.serialize(buf),
```

Comment:
IMO

```rust
Unset.serialize(buf)
```

looks nicer

Reply:
Wow, I didn't know this was possible.
Changed